### ISSUE TITLE
POP Avain Finland

### ISSUE THREAD

**Issue by tarikki** [PrivSec-dev/banking-apps-compat-report#481]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

POP Avain

### Link to app

https://play.google.com/store/apps/details?id=fi.poppankki.avain&hl=fi

### App version

2.2.3

### Country of the app

Finland

### Build Number

2024072800

### Device list

Pixel 8

### Profile app tested in

Owner profile

### Google Play installed?

Installed

### Where did you install this app from?

Google Play Store

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [X] Allowed
- [ ] Blocked

### Exploit protection compatibility mode

- [ ] Enabled
- [X] Disabled

### Memory tagging extension (MTE)

- [ ] Enabled
- [X] Disabled
- [ ] N/A

### Stock OS compatibility

- [X] Works
- [ ] Does not work
- [ ] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [X] N/A (Not supported by app)
- [ ] Not tested

### Description of the app's functionality

Digital ID/Key app of the POP Mobiili application. Works perfectly.

**Comment by Ekk0j** [comment]
Newer version of POP Avain (ver. 2.5.1) is not working with newest GrapheneOS. Sent message via customer service about the issue to POP Pankki (Bank). Waiting for reply.

**Comment by kobo28** [comment]
I contacted the bank about this issue. It was verified that GrapheneOS fails to meet API requirements demanded by the application. My guess is that they are using Google Integrity API to an extent that requires Google certified OS.

I shared [attestation-compatibility-guide](https://grapheneos.org/articles/attestation-compatibility-guide) but the response was that there will be no change. 

I am pressing the matter and making sure that the bank has made a conscious decision to only support Google certified OS.

**Comment by Ekk0j** [comment]
> I contacted the bank about this issue. It was verified that GrapheneOS fails to meet API requirements demanded by the application. My guess is that they are using Google Integrity API to an extent that requires Google certified OS.
> 
> I shared [attestation-compatibility-guide](https://grapheneos.org/articles/attestation-compatibility-guide) but the response was that there will be no change.
> 
> I am pressing the matter and making sure that the bank has made a conscious decision to only support Google certified OS.

Hello!

I also did that but i got no response from the customer service. Kinda sad that banks dont want privacy first.

**Comment by monorailsmoothie** [comment]
Could you please explain what exactly is not working for you? I've been using POP Avain for several years on different GrapheneOS devices. I installed it on my Pixel 6a on June 8, 2024. I'm using version 2.6.1.

**Comment by kobo28** [comment]
Apparently it is only the activation phase of the application that is blocked. I got a confirmation from customer service that my activation fails because GrapheneOS does not pass "the required API rules".

Everyone affected should contact the bank and bring this issue up. Their response to me was that no action will be taken...

**Comment by Ekk0j** [comment]
Its exactly the activation. Tried to change the rules but nothing helped. Maybe @monorailsmoothie have installed and logged in before the API rules were changed so it wont affect you.

**Comment by laukatti** [comment]
> Its exactly the activation. Tried to change the rules but nothing helped. Maybe [@monorailsmoothie](https://github.com/monorailsmoothie) have installed and logged in before the API rules were changed so it wont affect you.

I have it working, and I have installed it before API rules changed. I installed mine like 6 months ago.

**Comment by monorailsmoothie** [comment]
You're right. I did the activations several months ago. They're currently reforming their basic banking system, which should be finished by summer. Perhaps that's why they're reluctant to make major changes to the old system.

**Comment by kobo28** [comment]
The bank thanked me for the feedback but stated that the API checks are there by design. So only google signed Androids are supported and they indicated zero interest in changing the fact.

**Comment by monorailsmoothie** [comment]
I tried setting up POP Avain on a new device today with no luck. People using Säästöpankki Tunnistus are having similar problems. We should talk to Samlink because it's their platform.

**Comment by Ekk0j** [comment]
S-Mobiili (S-Pankki) does not have this issue. Heard that neither OP or Nordea. Maybe ill just do my banking with other device or just use browser.
